src/js/date.js

```diff
@@ -95,3 +96,58 @@ export const getNumericWeekday = (locale = 'en-uk', date = new Date()) => {
   });
   return weekdayIndex;
 };
+
+export const getLocaleDayMonthYear = (locale = 'en-uk', date = new Date()) => {
+  const objDate = date.toLocaleString(locale, { day: 'numeric', month: 'numeric', year: 'numeric' });
+  return objDate;
+};
+
+export const isDateValid = (locale = 'en-uk', inputValue = '') => {
+  let splittedInput = [];
+  let orderedInput = [];
+  let date = null;
+
+  if (locale === 'de-ch') {
```

Comment:
please check with indexOf if in the `inputValue` is more than one `.`

Reply:
I'm not sure whats the goal here?
Though I'm curious why I can't see usage native `Intl` APIs like [`Intl.DateTimeFormat`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/DateTimeFormat)

Reply:
Btw. don't think of using [`Date.parse`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Date/parse)

Reply:
Any valid Date will always be representable as a number, if it's invalid it's value will be `NaN`.
https://stackoverflow.com/questions/1353684/detecting-an-invalid-date-date-instance-in-javascript

Reply:
@NoahLiechti 
Cause the datepicker needs to be able to deal with various **internationalized**/**localized** formats.
And the [`Date.parse` API](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Date/parse) does not support this as far as I know.
Further more it's spec is far from being deterministic:
> It is not recommended to use Date.parse as until ES5, parsing of strings was entirely implementation dependent. There are still many differences in how different hosts parse date strings, therefore date strings should be manually parsed (a library can help if many different formats are to be accommodated).

`dataString` Parameter:
> A string representing an [RFC2822](https://tools.ietf.org/html/rfc2822#section-3.3) or (a variant of) ISO 8601 date (**other formats may be used, but results may be unexpected**).

For more reasons see above links

Reply:
Thank you @AndyOGo, we wont use date parse and find a solution that always works.